paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -323,17 +341,63 @@ def filter_aws_slaves(self, slaves_list):
         self.log.debug("IPs in AWS resources: {}".format(ips))
         slaves = [slave for slave in slaves_list if slave_pid_to_ip(slave['task_counts'].slave['pid']) in ips]
         slave_ips = [slave_pid_to_ip(slave['task_counts'].slave['pid']) for slave in slaves]
-        instance_descriptions = self.describe_instances([], region=self.resource['region'],
-                                                        instance_filters=[{'Name': 'private-ip-address',
-                                                                           'Values': slave_ips}])
+        instance_descriptions = self.describe_instances(
+            instances=[],
+            region=self.resource['region'],
+            instance_filters=[
+                {
+                    'Name': 'private-ip-address',
+                    'Values': slave_ips
+                }
+            ]
+        )
+        instance_statuses = self.describe_instance_status(
+            instances=[],
+            region=self.resource['region'],
+            instance_filters=[
+                {
+                    'Name': 'private-ip-address',
+                    'Values': slave_ips
+                }
+            ]
+        )
         instance_type_weights = self.get_instance_type_weights()
-        slave_instances = [PaastaAwsSlave(slave, instance_descriptions, instance_type_weights) for slave in slaves]
-        return slave_instances
+        paasta_aws_slaves = []
+        for slave in slaves:
```

Comment:
yes please. 
It kinda looks like there is one thing that correlates ips, one thing that correlates status's, and one thing that correlates descriptions?

And I think filter_aws_slaves is doing all three